<launch>
	<node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen">
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="framerate" value="10" />
		<param name="pixel_format" value="yuyv" />
		<param name="io_method" value="mmap" />
		<param name="exposure" value="5000" />
		<param name="autoexposure" value="false"/>
	</node>

	<node name="feature_extractor" type="feature_detector" pkg="visual_features_extractor" output="screen">
    <param name="orb_extractor" value="opencv"/>  		
    <param name="num_features" value="4000"/>
    <param name="scale_factor" value="1.2f"/>
    <param name="levels" value="8"/>
    <param name="ini_thrs" value="20"/>
    <param name="min_thrs" value="7"/>

		<param name="undistort_points" value="true"/>
		<param name="subscribe_to_state" value="true"/>
		<param name="publish_image_" value="true"/>		
		<remap from="~image_in" to="/usb_cam/image_raw"/>
                <remap from="~image_out" to="~debug"/>
		<remap from="~camera_info" to="/usb_cam/camera_info"/>
	</node>
    
</launch>
